sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -690,103 +688,80 @@ scatter_rgba(AccessorT acc, simd<uint32_t, N> offsets,
+                Op == __ESIMD_NS::atomic_op::load) {
+
+    if constexpr (!IsInt2BytePlus) {
+      static_assert(IsInt2BytePlus,
+                    "Integral 16-bit or wider type is expected");
     }
-    return true;
   }
-
-  // One source float operand.
-  if constexpr (Op == atomic_op::fmax || Op == atomic_op::fmin ||
-                Op == atomic_op::fadd || Op == atomic_op::fsub) {
-    if constexpr (NumSrc != 1) {
-      static_assert(NumSrc == 1, "One source operand is expected");
-      return false;
-    }
+  // FP ops (are always delegated to native::lsc::<Op>)
+  if constexpr (Op == __ESIMD_NS::atomic_op::fmax ||
+                Op == __ESIMD_NS::atomic_op::fmin ||
+                Op == __ESIMD_NS::atomic_op::fadd ||
+                Op == __ESIMD_NS::atomic_op::fsub) {
     if constexpr (!is_type<T, float, sycl::half>()) {
       static_assert((is_type<T, float, sycl::half>()),
                     "Type F or HF is expected");
-      return false;
     }
-    return true;
   }
-
-  // Two source operands.
-  if constexpr (Op == atomic_op::cmpxchg || Op == atomic_op::fcmpwr) {
-    if constexpr (NumSrc != 2) {
-      static_assert(NumSrc == 2, "Two source operands are expected");
-      return false;
+  if constexpr (Op == __ESIMD_NS::atomic_op::add ||
+                Op == __ESIMD_NS::atomic_op::sub ||
+                Op == __ESIMD_NS::atomic_op::min ||
+                Op == __ESIMD_NS::atomic_op::max ||
+                Op == __ESIMD_NS::atomic_op::bit_and ||
+                Op == __ESIMD_NS::atomic_op::bit_or ||
+                Op == __ESIMD_NS::atomic_op::bit_xor ||
+                Op == __ESIMD_NS::atomic_op::minsint ||
+                Op == __ESIMD_NS::atomic_op::maxsint) {
+    if constexpr (!IsInt2BytePlus) {
+      static_assert(IsInt2BytePlus,
+                    "Integral 16-bit or wider type is expected");
     }
-    if constexpr (Op == atomic_op::cmpxchg &&
-                  !is_type<T, uint16_t, uint32_t, uint64_t>()) {
-      static_assert((is_type<T, uint16_t, uint32_t, uint64_t>()),
-                    "Type UW, UD or UQ is expected");
-      return false;
-    }
-    if constexpr (Op == atomic_op::fcmpwr && !is_type<T, float, sycl::half>()) {
-      static_assert((is_type<T, float, sycl::half>()),
-                    "Type F or HF is expected");
-      return false;
+    constexpr bool IsSignedMinmax = (Op == __ESIMD_NS::atomic_op::minsint) ||
+                                    (Op == __ESIMD_NS::atomic_op::maxsint);
+    constexpr bool IsUnsignedMinmax = (Op == __ESIMD_NS::atomic_op::min) ||
+                                      (Op == __ESIMD_NS::atomic_op::max);
+
+    if constexpr (IsSignedMinmax || IsUnsignedMinmax) {
+      constexpr bool SignOK = std::is_signed_v<T> == IsSignedMinmax;
+
+      if constexpr (!SignOK) {
+        static_assert(SignOK, "Signed/unsigned integer type expected for "
+                              "signed/unsigned min/max operation");
+      }
```

Comment:
This looks little bit odd to me. Can it be just a static_assert?
```suggestion
      static_assert(SignOK, "Signed/unsigned integer type expected for "
                            "signed/unsigned min/max operation");
```

Reply:
If you decide to do changes for this place, then please do similar changes in several other places modified by this PR.